src/settings.rs

```diff
@@ -34,8 +33,7 @@ pub const DETERMINATE_NIXD_BINARY: Option<&[u8]> =
 
 #[cfg(not(all(
     feature = "determinate-nix",
-    target_os = "linux",
-    not(target_arch = "x86")
+    not(all(target_os = "linux", target_arch = "x86"))
```

Comment:
Same here: "If determinate-nix is not enabled, or the target is i686-linux, do not embed the binary"